welltestpy/tools/plotter.py

```diff
@@ -895,3 +895,44 @@ def plotsensitivity(
                 bbox_inches="tight",
             )
     return ax
+
+def diagnostic_plot_pump_test(observation, derivative):
+    """plot the derivative with the original data.
+
+               Parameters
+               ----------
+               observation : :class:`welltestpy.data.Observation`
+                    The observation to calculate the derivative.
+
+               derivative : :class:'array'
+                    An array with the calculated derivative
+
+                Returns
+                ---------
+                Diagnostic plot
+          """
+    head, time = observation()
+    head = np.array(head, dtype=float).reshape(-1)
+    time = np.array(time, dtype=float).reshape(-1)
+    # setting variables
+    x = time
+    y = head
+    sx = time
+    sy = head
+    dx = time
+    dy = derivative
+
+    # plotting
+    fig, ax = plt.subplots(dpi=75, figsize=[7.5, 5.5])
```

Comment:
If you take a look at the other plotting routines, you see, that they share a certain signature, including `fig` and `ax` (when plot includes a single axis) to be passed to the routine. Please follow this. You could also try to imitate the `style="WTP"` to select the plotting style. Have a look at this for example:
https://github.com/GeoStat-Framework/welltestpy/blob/5bdf033b7dbf3439862424ff23ba3f7b234d150d/welltestpy/tools/plotter.py#L862